docker/wc-admin-docker-dev/README.md

```diff
@@ -0,0 +1,111 @@
+# WooCommerce Docker Setup with WP-ENV
+
+Docker development setup for WooCommerce Admin with WP-ENV.
+
+## Prerequisites
+
+Please install WP-ENV before getting started. You can find more about WP-ENV on [here](https://github.com/WordPress/gutenberg/tree/master/packages/env).
+
+The following command installs WP-ENV globally.
+
+`npm -g i @wordpress/env`
+
+
+## Getting Started
+
+1. Navigate to the root of WooCommerce Admin source code.
+2. Start the docker container by running `wp-env start`
+
+You should see the following output
+
+```
+WordPress development site started at http://localhost:8888/
+WordPress test site started at http://localhost:8889/
+MySQL is listening on port 55003
+```
+
+The port # might be different depending on your `.wp-env.override.json` configuration.
+
+## Overriding the Default Configuration
+
+The default configuration comes with PHP 7.4, WooCommerce 5.0, and a few WordPress config values.
+
+You can create `.wp-env.override.json` file and override the default configuration values.
+
+You can find more about `.wp-env.override.json` configuration [here](https://github.com/WordPress/gutenberg/tree/master/packages/env#wp-envoverridejson).
+
+**Example: Overriding PHP version to 8.0**
+
+Create `.wp-env.override.json` in the root directory with the following content.
+
+
+```json
+{
+	"phpVersion": "8.0"
+}
+```
+
+Once you have the file, restart the container with `wp-env restart` command.
+
+## Getting Started with Developing
+
+The current WP-ENV does not support creating a custom `Dockerfile`. Due to the limitation, we need to install node and composer manually. There is a [draft PR](https://github.com/WordPress/gutenberg/pull/26047) to improve the limitation. You need to run this command every time you stop and start the container, but this won't take a minute.
+
+`npm run docker-dev -- install`
+
+
+### Using Docker for Backend Only
+
+Since you already have node on your client, you can use WC Admin Docker setup just for the backend.
+
+Run the following commands in your client.
+
+1. `npm install`
+2. `npm run dev`
+
+Run the following command to install composer packages for the backend (PHP).
+
+`npm run docker-dev -- composer-install`
+
+
+### Using Docker for Both Backend and Frontend
+
+Run `npm run docker-dev -- setup`.
```

Comment:
You're correct. WP-ENV mounts the plugin directory.

You can also run the composer and npm locally, but not everyone has a composer available locally. Since this docker setup is intended to work as it is without anything from the user's local environment, I think we should have those commands for those without npm and composer.

Reply:
We can also remove NPM related command completely and ask users to run them locally, but I think composer command should stay as it's harder to setup (Installing and switching versions) PHP environment than node.

Reply:
@moon0326 I don't think this makes sense. People should have those available locally and running them in the container is a layer of indirection that makes no sense.

Reply:
As another argument if you *really* wanted to install composer and npm for them, then you *still* don't need to do it in the container because its a bind mount, you'd just have a script to do it for them locally.

Reply:
@samueljseay The main goal of having a Docker setup is to have the exact same environment (Linux distro, PHP version, PHP settings, and so on) and minimize local setup in my opinion. It's pretty easy to install Node on Mac OS, Linux, and Windows. The problem is PHP and Composer.

Mac OS comes with PHP installed by default, but Linux and Windows don't. Installing PHP can be pretty tricky depending on local configuration.

With that said, I totally understand your concern and point. If we remove NPM and Composer installation from the container, things get much simpler. I would be happier, to be honest. 

I'm going to remove them and update the doc. If anyone complains about them later, I think we can add them again.

Thank you for the input! Updates are coming very soon 👍

Reply:
> If we remove NPM and Composer installation from the container, things get much simpler. I would be happier, to be honest.

That sounds good. Keep it simple 👍 

As a wp-env user, I just use npm and composer locally. However, wp-env does have a composer service out of the box. I think that could be used instead if folks don't want a local composer install.

https://github.com/WordPress/gutenberg/blob/8ab2453b7afffcee9f383d668399778597c4e181/packages/env/lib/build-docker-compose-config.js#L214-L216

Reply:
@adrianduffell Thank you!! Good find!!! 👍